src/components/Pf/PfTitle.tsx

```diff
@@ -0,0 +1,110 @@
+import * as React from 'react';
+import MissingSidecar from '../../components/MissingSidecar/MissingSidecar';
+import { Link } from 'react-router-dom';
+import { ServiceIcon, BundleIcon, ApplicationsIcon } from '@patternfly/react-icons';
+import { CytoscapeGraphSelectorBuilder } from '../CytoscapeGraph/CytoscapeGraphSelector';
+import { style } from 'typestyle';
+import { NodeType } from '../../types/Graph';
+
+const PfTitleStyle = style({
+  fontSize: '19px',
+  fontWeight: 400,
+  margin: '20px 0',
+  padding: '0'
+});
+
+interface PfTitleProps {
+  location?: {
+    pathname: string;
+    search: string;
+  };
+  istio?: boolean;
+  hideGraphLink?: boolean;
+}
+
+interface PfTitleState {
+  type: string;
+  namespace: string;
+  name: string;
+  cytoscapeGraph: string;
+  graphType: string;
+  icon: JSX.Element;
+}
+
+class PfTitle extends React.Component<PfTitleProps, PfTitleState> {
+  constructor(props: PfTitleProps) {
+    super(props);
+    const namespaceRegex = /namespaces\/([a-z0-9\-]+)\/([a-z0-9\-]+)\/([a-z0-9\-]+)(\/([a-z0-9\-]+))?(\/([a-z0-9\-]+))?/;
```

Comment:
Note that namespaces might have '.' character
https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
I learnt that in the hard way,
I used a regexp (actually, this was recommended by k8s) in
https://github.com/kiali/kiali-ui/blob/master/src/components/IstioWizards/ThreeScaleIntegration.tsx#L62